### ISSUE TITLE
Support for OpenTelemetry traces and span/trace ids in log lines

### ISSUE THREAD

**Issue by gburd** [dagster-io/dagster#12353]
### What's the use case?

Dagster's executions result in nice trace-like structure well visualized in your GUI, those same traces would be useful in other ways were they to be published as OpenTelemetry tracing events.  To connect all the dots when this is enabled all log lines should include trace and span-id so they can be correlated to the traces.  Many companies are adopting OpenTracing to debug issues, we at Klar require this for all services.  This is a gap in the "observability triangle" for Dagster that could easily be plugged.

### Ideas of implementation

Possibly useful https://gist.github.com/gburd/01d0d4214ab4e4fb82c7fe0179f4f60b and [Implement a default telemetry provider parallel to the default logger · Issue #11191 · dagster-io/dagster](https://github.com/dagster-io/dagster/issues/11191)

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by kuzaxak** [comment]
Hey, any updates on that? It would be nice to have this feature.

**Comment by anentropic** [comment]
I came looking for a NewRelic integration

Noting this comment: https://github.com/dagster-io/dagster/discussions/7221#discussioncomment-2461420 ... Which points out that NewRelic supports OpenTelemetry traces 

https://docs.newrelic.com/docs/opentelemetry/opentelemetry-introduction/

So having OT support would cover multiple bases

**Comment by pasunboneleve** [comment]
If you already have an **OpenTelemetry** Collector running in your infrastructure, then you already have that feature. The **OpenTelemetry** Collector [can read Prometheus streams](https://opentelemetry.io/blog/2024/prom-and-otel/), so you can just point [dagster-prometheus](https://dagster.io/integrations/dagster-prometheus) to it and from the **OpenTelemetry** Collector you can send the streams to Honeycomb or whatever other vendor you like.

I didn't implement it myself (yet), but I intend to!

**Comment by kuzaxak** [comment]
@dmvianna Prometheus exposes metrics, not traces

**Comment by pasunboneleve** [comment]
@kuzaxak good point. I ignored traces because other vendor software I use (**Airbyte** specifically) suffer from the same problem. I don't think many of these products do provide traces. If you know of any, I would be first to adopt!

**Comment by luck0r** [comment]
would be very useful to have traces to find Bottlenecks

**Comment by HirofumiTsuda** [comment]
This is exactly the gap I ran into too. Built `dagster_otel` to plug it: a decorator (`@traced()`) that stacks under `@op`/`@asset` (no monkeypatching, no giving up ownership of the op/asset definition to a wrapping framework), propagates trace context across `multiprocess`/`k8s_job_executor` steps via Dagster's own run storage (since each step is usually its own process), and forwards trace_id/span_id onto `context.log` so log lines correlate with the trace.

Public here if useful: https://github.com/HirofumiTsuda/dagster-otel

Doesn't solve the broader "default telemetry provider" ask in #11191, just this specific traces + log-correlation piece.
